pandas/core/arrays/boolean.py

```diff
@@ -372,34 +372,22 @@ def astype(self, dtype, copy: bool = True) -> ArrayLike:
             if incompatible type with an BooleanDtype, equivalent of same_kind
             casting
         """
-        from pandas.core.arrays.string_ import StringDtype
-
         dtype = pandas_dtype(dtype)
 
-        if isinstance(dtype, BooleanDtype):
-            values, mask = coerce_to_array(self, copy=copy)
-            if not copy:
-                return self
-            else:
-                return BooleanArray(values, mask, copy=False)
-        elif isinstance(dtype, StringDtype):
-            return dtype.construct_array_type()._from_sequence(self, copy=False)
+        if is_extension_array_dtype(dtype):
```

Comment:
small consistency nit, in the files below you use `if isinstance(dtype, ExtensionDtype)` for this, so should probably use one pattern

Reply:
sure, will update